cmake/cmake-utilities.cmake

```diff
@@ -126,4 +126,13 @@ macro(assign_bool variable)
      else()
          set(${variable} OFF)
      endif()
-endmacro()
\ No newline at end of file
+endmacro()
+
+# add a csound test based on a .csd file
+function(add_commandline_test filename status)
```

Comment:
I think the name `add_commandline_test ` is not very clear what is the purpose. The command line is immaterial here as most things are done via the CLI. It is really adding an integration test. I think `add_integration_test` is better.